### ISSUE TITLE
Basic video editor

### ISSUE THREAD

**Issue by PhieF** [Chocobozzz/PeerTube#1029]
Hi

on a very very low priority it would be nice to have a basic video editor

I would see it like that : 
after uploading a file, user will be asked if he wants to edit his video. If he does, then he will be warned that a transcoding job needs to be done before (only to the best resolution). Once it is done, a basic editor to cut video and add a sound channel, perhaps some filters I don't know (doesn't seem to be important) preview for cut / sound can be done in the browser, I guess.
then after editing the publish page is displayed, then the real edit is rendered with ffmpeg on the server with cut parameters and audiofile, usual transcoding jobs begin, and that's it

**Comment by PhieF** [comment]
Well I guess there are more urgent things to do, indeed :P perhaps with the future extension system, it would make more sense

**Comment by benrob0329** [comment]
I feel like there are so many FOSS editors out there now thats its rather unnecessary to put the extra load on the server IMHO.

**Comment by faissaloo** [comment]
Use Kdenlive, video editing in the browser is not something that can be reasonably done.

**Comment by strypey** [comment]
> video editing in the browser is not something that can be reasonably done.

I'm not sure that's true, as there are a *lot* of projects writing software to do just this. One that I know has a free code community edition is Kaltura:
https://www.kaltura.org/

That said, I agree that video-editing is a specialist and resource-hungry activity for a server to handle. I think it would make more sense to develop a simple, web-based editor that sysadmins could choose to run alongside a PT instance, maybe integrated with the PT UI via a plug-in? Perhaps it could be developed as a soft fork of something like Kaltura, stripping it down to a basic set of commonly-used functions (eg trim off each end, add soundtrack), so it can doesn't use the resources of a full Kaltura stack?

**Comment by joshmorel** [comment]
I think this would expand the scope of the project into dangerous territory. Focusing effort on something that can be done much better with other tools would take away from the special thing PeerTube is trying to do. 

Worthwhile discussion item however!

**Comment by Yetangitu** [comment]
In #1399 I suggested adding basic editing options which led me here. I think the feature suggested here is far too heavy for PeerTube and as such not feasible or desirable. What can be added is basic editing which can be achieved by manipulating metadata (e.g. rotation) without requiring transcoding. These operations are fast and easy on the server and can be achieved using the same tool (_ffmpeg_) as used for transcoding.

**Comment by test2a** [comment]
@Chocobozzz would it be possible to integrate already existing programs that do exactly this now with wasm and ffmpeg? https://github.com/antoineMoPa/cutvid.io this is a AGPL 3 licensed video editor that runs in the browser. The data would come from the server storage and editing would happen in browser.
